### PR TITLE
cni: add calico/canal as alternate to flannel and cilium plugin

### PR DESCRIPTION
**Pull request by esimone74**
Canal allow users to easily deploy Calico and flannel networking together as a unified networking solution to provide network policy enforcement with the flannel overlay and non-overlay network connectivity options.

### CODE REVIEW

File: pillar/cni.sls [hunk 1]
```diff
@@ -17,6 +17,11 @@ flannel:
 # 8 - Display HTTP request contents.
   log_level:      '2'
 
+# the calico backend
+calico:
+  node: 'quay.io/calico/node:v2.6.9'
```
Comment (inercia) [inline pillar/cni.sls]: I think we do not support images based on things other than `sles` (but @flavio should confirm this).

File: salt/cni/canal.yaml.jinja [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+        },
+        {
+          "type": "portmap",
+          "capabilities": {"portMappings": true},
+          "snat": true
+        }
+      ]
+    }
+
+  net-conf.json: |
```
Comment (inercia) [inline salt/cni/canal.yaml.jinja]: This is for _Flannel_, it wouldn't be necessary for _Calico_.

**Comment by esimone74** [comment]
Hi,

> On 18 May 2018, at 09:46, Alvaro <notifications@github.com> wrote:
> 
> @inercia requested changes on this pull request.
> 
> In pillar/cni.sls <https://github.com/kubic-project/salt/pull/547#discussion_r189186931>:
> 
> +  node: 'quay.io/calico/node:v2.6.9'
> I think we do not support images based on things other than sles (but @flavio <https://github.com/flavio> should confirm this).
> 
> 
We’ll wait for him.

> 
> In salt/cni/canal.yaml.jinja <https://github.com/kubic-project/salt/pull/547#discussion_r189187218>:
> 
> > +            "k8s_auth_token": "__SERVICEACCOUNT_TOKEN__"
> +          },
> +            "kubernetes": {
> +            "k8s_api_root": "https://__KUBERNETES_SERVICE_HOST__:__KUBERNETES_SERVICE_PORT__",
> +            "kubeconfig": "__KUBECONFIG_FILEPATH__"
> +          }
> +        },
> +        {
> +          "type": "portmap",
> +          "capabilities": {"portMappings": true},
> +          "snat": true
> +        }
> +      ]
> +    }
> +
> +  net-conf.json: |
> This is for Flannel, it wouldn't be necessary for Calico.
> 
Calico/Canal use Flannel for networking, that’s why it is still present[1].
In the DaemonSet, the SUSE Flannel image is used to provide networking.

In a second time I’ll work on a pure Calico implementation.

[1] https://docs.projectcalico.org/v2.6/getting-started/kubernetes/installation/hosted/canal/ <https://docs.projectcalico.org/v2.6/getting-started/kubernetes/installation/hosted/canal/>

Best regards,
Ettore Simone

File: salt/cni/canal.yaml.jinja (hunk 2 above)
Comment (esimone74) [reply salt/cni/canal.yaml.jinja]: Calico/Canal use Flannel as networking layer.

**Comment by flavio** [comment]
@esimone74 thanks for the PR, we are going to discuss internally how to handle that.

File: pillar/cni.sls (hunk 1 above)
Comment (mjura) [reply pillar/cni.sls]: It is fine as long, we keep this as empty pillar values

File: pillar/cni.sls [hunk 3]
```diff
@@ -17,6 +17,11 @@ flannel:
...
+  cni:  'quay.io/calico/cni:v1.11.5'
```
Comment (mjura) [inline pillar/cni.sls]: We keep it as empty pillar values

**Comment by MalloZup** [comment]
@flavio @inercia @mjura  closing. feel free to open a new pr if needed. thank you for the time

**Pull request closed without merging**